src/tests/ngraph_helpers/snippets_ngraph_functions/include/two_element_wise_function.hpp

```diff
@@ -0,0 +1,156 @@
+        constructor_validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        set_output_type(
+            0,
+            output_type == element::undefined ? get_input_element_type(0) : output_type,
+            get_input_partial_shape(0));
+    }
+
+    element::Type get_output_type() const { return output_type; }
+
+protected:
+    element::Type output_type;
+};
+
+class DummyOperation1 : public BaseDummyOperation {
+public:
+    OPENVINO_OP("DummyOperation1", "test::snippets");
+
+    DummyOperation1(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : BaseDummyOperation(arg0, arg1, output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyOperation1(const BaseDummyOperation& op) : DummyOperation1(op.get_input_source_output(0), op.get_input_source_output(1), op.get_output_type()) {
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation1>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+class DummyOperation2 : public BaseDummyOperation {
+public:
+    OPENVINO_OP("DummyOperation2", "test::snippets");
+
+    DummyOperation2(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : BaseDummyOperation(arg0, arg1, output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyOperation2(const BaseDummyOperation& op) : DummyOperation2(op.get_input_source_output(0), op.get_input_source_output(1), op.get_output_type()) {
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation2>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+class TwoElementWiseFunctionParams {
+public:
+    class Actual {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+        element::Type convertion_after_op2;
+    };
+
+    class Expected {
+    public:
+        std::pair<element::Type, element::Type> convertion_before_op1;
+        element::Type convertion_before_op2_1;
+        std::pair<element::Type, element::Type> convertion_before_op2_2;
+        element::Type convertion_after_op2;
+        element::Type convertion_before_result;
+    };
+};
+
+/**
+ * @class TwoElementWiseFunction
+ * @brief The function for unit tests only, includes dummy operations.
+ */
+class TwoElementWiseFunction : public SnippetsFunctionBase {
+public:
+    explicit TwoElementWiseFunction(
```

Comment:
Minor: `Elementwise` is spelled in one word without spaces everywhere else (not `Element Wise`), let's align. The same is true for the file name.

Reply:
not actual more